pandas/io/data.py

```diff
@@ -698,8 +698,18 @@ def _option_frames_from_url(self, url):
 
     def _get_underlying_price(self, url):
         root = self._parse_url(url)
-        underlying_price = float(root.xpath('.//*[@class="time_rtq_ticker Fz-30 Fw-b"]')[0]\
-            .getchildren()[0].text)
+        underlying_price = root.xpath('.//*[@class="time_rtq_ticker Fz-30 Fw-b"]')[0]\
+            .getchildren()[0].text
+
+        try:
+            underlying_price = float(underlying_price)
+        except ValueError:
+            # see if there is a comma thousands separator that needs to be filtered out
+            underlying_price = ''.join(c for c in underlying_price if c != ',')
```

Comment:
I still think this should just be `underlying_price.replace(',', '')`. As noted in the other PR, this method is not deprecated in Python 3: https://docs.python.org/3/library/stdtypes.html#str.replace